### PR TITLE
added complete Qwen2.5-7B tensor parallel bounty implementation

### PR DESCRIPTION
**Pull request by ekulkisnek**
### Ticket
[Link to Github Issue
](https://github.com/tenstorrent/tt-forge/issues/52)

### Problem description
Includes a JAX tensor parallel implementation of Qwen2.5-7B for the mentioned bounty. With deterministic settings it provides token for token identical outputs to a pytorch implementation. 

### What's changed
Directly compared logits each step between a pytorch implementation and a single device JAX implementation. Once the single device JAX worked, I used shard_map to get it parallel. Tested for 1x8. 

### Checklist
- [x] Working tensor parallel JAX Qwen2.5-7B model with same GSM8K performance as a pytorch implementation.

### CODE REVIEW

**Comment by marty1885** [comment]
@ekulkisnek Hi!been a while since the PR is last updated. How is it going? Any blockers we can help resolve to help you get the PR merged?
